From 9a8e4b73779add04a9852759c857ef082b19cf44 Mon Sep 17 00:00:00 2001
From: Apollon Oikonomopoulos <apoikos@gmail.com>
Date: Wed, 24 Jul 2013 09:22:00 +0300
Subject: [PATCH] daemon-util: pass --oknodo at rotate_logs

daemon-util's rotate_logs() did not pass --oknodo to start-stop-daemon while
HUPing the daemon processes. As a result, rotate_logs would fail for a
non-running daemon causing rotate_all_logs to exit prematurely.

We can fix this by either passing --oknodo to start-stop-daemon, or have
rotate_all_logs ignore any errors. The former is probably a safer approach - as
it doesn't mask any potential coding errors in rotate_logs - and is also
semantically correct (we only care about log rotation if a daemon is actually
running).

Signed-off-by: Apollon Oikonomopoulos <apoikos@gmail.com>
Reviewed-by: Guido Trotter <ultrotter@google.com>
---
 daemons/daemon-util.in |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/daemons/daemon-util.in b/daemons/daemon-util.in
index 2be14e9..848a122 100644
--- a/daemons/daemon-util.in
+++ b/daemons/daemon-util.in
@@ -324,7 +324,7 @@ rotate_logs() {
 
   if type -p start-stop-daemon >/dev/null; then
     start-stop-daemon --stop --signal HUP --quiet \
-      --pidfile $pidfile
+      --oknodo --pidfile $pidfile
   else
     _ignore_error killproc \
       -p $pidfile \
-- 
1.7.10.4

